src/main/java/org/snomed/aag/data/domain/CriteriaItemSignOffFactory.java

```diff
@@ -0,0 +1,32 @@
+package org.snomed.aag.data.domain;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class CriteriaItemSignOffFactory {
+	/**
+	 * Return CriteriaItemSignOff. If branch is the same as ProjectAcceptanceCriteria.branchPath, CriteriaItemSignOffProject will be returned. Otherwise, CriteriaItemSignOffTask
+	 * will be returned.
+	 *
+	 * @param criteriaItemId            Value for CriteriaItemSignOff.
+	 * @param branch                    Value for CriteriaItemSignOff.
+	 * @param branchHeadTimestamp       Value for CriteriaItemSignOff.
+	 * @param projectIteration          Optional value for CriteriaItemSignOff.
+	 * @param userId                    Value for CriteriaItemSignOff.
+	 * @param projectAcceptanceCriteria Required for determining type of CriteriaItemSignOff
+	 * @return CriteriaItemSignOff
+	 */
+	public CriteriaItemSignOff create(String criteriaItemId, String branch, Long branchHeadTimestamp, Integer projectIteration, String userId, ProjectAcceptanceCriteria projectAcceptanceCriteria) {
+		boolean branchProjectLevel = projectAcceptanceCriteria.isBranchProjectLevel(branch);
+		if (branchProjectLevel) {
```

Comment:
This boolean could be used inline rather than assigned to a variable.

Reply:
I have done this.